src/common/misc.c

```diff
@@ -1071,3 +1072,44 @@ daos_prop_copy(daos_prop_t *prop_req, daos_prop_t *prop_reply)
 	}
 	return rc;
 }
+
+static __thread struct {
+	uint64_t	host_id;
+	uint64_t	hlc;
+} dtx_id_tls;
+
+/* for the memcpy in daos_dti_gen */
+D_CASSERT(sizeof(((struct dtx_id *)0)->dti_uuid) >= sizeof(dtx_id_tls));
+
+static uint64_t
+daos_gethostid(void)
+{
+#define HOSTNAME_LEN 1024
+	char	hostname[HOSTNAME_LEN] = { 0 };
+	int	rc;
+
+	rc = gethostname(hostname, HOSTNAME_LEN);
+	if (rc == 0)
+		return d_hash_murmur64((unsigned char *)hostname, HOSTNAME_LEN,
+				       609815U);
+
+	D_WARN("gethostname failed rc %d (errno %d).\n", rc, errno);
+	return (gethostid() << 32) | rand();
```

Comment:
do we seed rand() somewhere?   Otherwise, it is "automatically seeded with a value of 1" which means every host will get same value.

Reply:
I remember seed it somewhere, but checked it is not true. just feed a seed in init.